### PR TITLE
Cleanup deployment

### PR DESCRIPTION
**Pull request by garagatyi**
## What does this PR do?
This PR is designed to be reviewed in a commit-by-commit manner. This simplifies diffs of commits significantly and doesn't shuffle them.

Simplify Che deployment configuration by:
- Renaming ConfigMap entries to match actual env vars they are supposed to refer
- Injecting all the ConfigMap entries as bunch instead of adding them one-by-one in the deployment
- Moving some env vars hardcoded in the deployment to the config map
- Stacking all the env vars imports from secrets together
- Doing similar stuff for minishift addon
- Removing not needed property (see [this diff](https://github.com/redhat-developer/rh-che/commit/d9d67c4125e5184f99c4df09f73de0ed1899f173#diff-c98f3dce05b630b31da8dbdfee24da1fR193))  

### What issues does this PR fix or reference?
Inspired by https://github.com/eclipse/che/pull/12289

### CODE REVIEW

**Comment by garagatyi** [comment]
Marked as WIP since we will still need to deal with HK issue after we agreed to merge it

**Comment by garagatyi** [comment]
@l0rd do you still want to be a code owner here?

**Comment by garagatyi** [comment]
[test]

**Comment by garagatyi** [comment]
[test]

**Comment by garagatyi** [comment]
@amisevsk we could also add secret reference instead of duplicating it in the deployment. But I would leave it for later improvements.

@ibuziuk @davidfestal @rhopp @Katka92 @tdancs WDYT?

**Comment by garagatyi** [comment]
@ibuziuk @davidfestal @rhopp @Katka92 @tdancs Please, review

**Comment by garagatyi** [comment]
@ibuziuk do we need to update prod-preview config map by SD team too?

**Comment by garagatyi** [comment]
see https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/2501

**Comment by ibuziuk** [comment]
> do we need to update prod-preview config map by SD team too?

@garagatyi yes

**Comment by garagatyi** [comment]
@ibuziuk keep forgetting =(

**Comment by garagatyi** [comment]
@ibuziuk https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/2501 is applied to prod and prod-preview. Can we proceed?

File: dev-scripts/deploy_custom_rh-che.sh [hunk 1]
```diff
@@ -295,27 +295,26 @@ else
...
-                      .\"data\".\"che-api\" = \"http$SECURE://rhche-$RH_CHE_PROJECT_NAMESPACE.devtools-dev.ext.devshift.net/api\" |
-                      .\"data\".\"che-websocket-endpoint\" = \"ws$SECURE://rhche-$RH_CHE_PROJECT_NAMESPACE.devtools-dev.ext.devshift.net/api/websocket\" |
+                  yq ".\"data\".\"CHE_HOST\" = \"rhche-$RH_CHE_PROJECT_NAMESPACE.devtools-dev.ext.devshift.net\" |
+                      .\"data\".\"CHE_INFRA_KUBERNETES_BOOTSTRAPPER_BINARY__URL\" = \"http$SECURE://rhche-$RH_CHE_PROJECT_NAMESPACE.devtools-dev.ext.devshift.net/agent-binaries/linux_amd64/bootstrapper/bootstrapper\" |
+                      .\"data\".\"CHE_API\" = \"http$SECURE://rhche-$RH_CHE_PROJECT_NAMESPACE.devtools-dev.ext.devshift.net/api\" |
+                      .\"data\".\"CHE_WEBSOCKET_ENDPOINT\" = \"ws$SECURE://rhche-$RH_CHE_PROJECT_NAMESPACE.devtools-dev.ext.devshift.net/api/websocket\" |
                       .\"metadata\".\"name\" = \"rhche\" |
-                      .\"data\".\"che-openshift-secure-routes\" = \"$RH_CHE_USE_TLS\" |
```
Comment (ibuziuk) [inline dev-scripts/deploy_custom_rh-che.sh]: @garagatyi could you clarify why TLS options are not needed in the script ?

Comment (garagatyi) [reply dev-scripts/deploy_custom_rh-che.sh]: It is replaced with CHE_INFRA_OPENSHIFT_TLS__ENABLED

Comment (ibuziuk) [reply dev-scripts/deploy_custom_rh-che.sh]: hmm.. I got a bit lost why we had two properties "che-openshift-secure-routes" & "che-secure-external-urls"

Comment (ibuziuk) [reply dev-scripts/deploy_custom_rh-che.sh]: funny, looks like `che-secure-external-urls` is not used at all currently in the rh-che template

**Comment by ibuziuk** [comment]
@garagatyi great job. +1 to apply

File: dev-scripts/deploy_custom_rh-che.sh (hunk 1 above)
Comment (garagatyi) [reply dev-scripts/deploy_custom_rh-che.sh]: yes, it is unused

**Pull request merged**